### PR TITLE
Fix mlflow platform argument

### PR DESCRIPTION
**Pull request by WeichenXu123**
Signed-off-by: Weichen Xu <weichen.xu@databricks.com>

## What changes are proposed in this pull request?

Fix mlflow platform argument

## How is this patch tested?

N/A

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [x] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/models/docker_utils.py [hunk 1]
```diff
@@ -106,19 +107,26 @@ def _build_image(image_name, entrypoint, mlflow_home=None, custom_setup_steps_ho
...
-            [
-                "docker",
-                "build",
-                "-t",
-                image_name,
-                "-f",
-                "Dockerfile",
+
+        client = docker.from_env()
+        should_add_platform = int(client.version()["Version"].split(".")[0]) > 18
```
Comment (harupy) [inline mlflow/models/docker_utils.py:112]: Can you insert a comment here?

File: mlflow/models/docker_utils.py [hunk 2]
```diff
@@ -106,19 +107,26 @@ def _build_image(image_name, entrypoint, mlflow_home=None, custom_setup_steps_ho
...
+            commands.extend([
                 # Enforcing the AMD64 architecture build for Apple M1 users
                 "--platform",
                 "linux/amd64",
-                ".",
-            ],
+            ])
+        commands.append(".")
+        proc = Popen(
+            commands,
```
Comment (harupy) [inline mlflow/models/docker_utils.py:129]: Can we refactor this to below?

```python
        client = docker.from_env()
        # `--platform` option for `docker build` is only available in Docker > 18
        is_platform_supported = int(client.version()["Version"].split(".")[0]) > 18
        # Enforcing the AMD64 architecture build for Apple M1 users
        platform_option = ["--platform", "linux/amd64"] if is_platform_supported else []
        commands = [
            "docker",
            "build",
            "-t",
            image_name,
            "-f",
            "Dockerfile",
            *platform_option,
            ".",
        ]
```

File: mlflow/models/docker_utils.py [hunk 3]
```diff
@@ -106,24 +108,21 @@ def _build_image(image_name, entrypoint, mlflow_home=None, custom_setup_steps_ho
...
-            ],
-            cwd=cwd,
-            stdout=PIPE,
-            stderr=STDOUT,
-            universal_newlines=True,
-        )
+
+        client = docker.from_env()
+        is_platform_supported = int(client.version()["Version"].split(".")[0]) > 18
+        platform_option = ["--platform", "linux/amd64"] if is_platform_supported else []
```
Comment (harupy) [inline mlflow/models/docker_utils.py:114]: @WeichenXu123 Why did you remove the comments?

File: mlflow/models/docker_utils.py [hunk 4]
```diff
@@ -106,24 +108,23 @@ def _build_image(image_name, entrypoint, mlflow_home=None, custom_setup_steps_ho
...
-            ],
-            cwd=cwd,
-            stdout=PIPE,
-            stderr=STDOUT,
-            universal_newlines=True,
-        )
+
+        client = docker.from_env()
+        # In Docker < 19, `docker build` doesn't support the `--platform` option
+        is_platform_supported = int(client.version()["Version"].split(".")[0]) > 18
```
Comment (harupy) [inline mlflow/models/docker_utils.py:114]:
```suggestion
        is_platform_supported = int(client.version()["Version"].split(".")[0]) >= 19
```

nit

**Review by harupy — APPROVED** [review]
@WeichenXu123 Pushed one commit to fix a nit. LGTM!

**Comment by WeichenXu123** [comment]
@harupy Thanks!

**Comment by tugbayatilla** [comment]
@WeichenXu123 may I ask the decision behind "# Enforcing the AMD64 architecture build for Apple M1 users" ?
